AdminRouter/start-windows-build.ps1

```diff
@@ -0,0 +1,266 @@
+Param(
+    [Parameter(Mandatory=$false)]
+    [string]$GitURL="https://github.com/apache/httpd", 
+    [Parameter(Mandatory=$false)]
+    [string]$Branch="master",
+    [Parameter(Mandatory=$false)]
+    [string]$CommitID,
+    [Parameter(Mandatory=$false)]
+    [string]$ParametersFile="${env:WORKSPACE}\build-parameters.json"
+)
+
+$ErrorActionPreference = "Stop"
+
+$globalVariables = (Resolve-Path "$PSScriptRoot\..\global-variables.ps1").Path
+$ciUtils = (Resolve-Path "$PSScriptRoot\..\Modules\CIUtils").Path
+
+Import-Module $ciUtils
+. $globalVariables
+
+$global:PARAMETERS = @{
+    "BUILD_STATUS" = $null
+    "LOGS_URLS" = @()
+    "FAILED_COMMAND" = $null
+}
+
+function Install-Prerequisites {
+    $prerequisites = @{
+        'git'= @{
+            'url'= $GIT_URL
+            'install_args' = @("/SILENT")
+            'install_dir' = $GIT_DIR
+        }
+        '7zip'= @{
+            'url'= $7ZIP_URL
+            'install_args'= @("/q")
+            'install_dir'= $7ZIP_DIR
+        }
+    }
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    foreach($program in $prerequisites.Keys) {
+        if(Test-Path $prerequisites[$program]['install_dir']) {
+            Write-Output "$program is already installed"
+            continue
+        }
+        Write-Output "Downloading $program from $($prerequisites[$program]['url'])"
+        $fileName = $prerequisites[$program]['url'].Split('/')[-1]
+        $programFile = Join-Path $env:TEMP $fileName
+        Start-ExecuteWithRetry { Invoke-WebRequest -UseBasicParsing -Uri $prerequisites[$program]['url'] -OutFile $programFile} $RetryInterval 20
+        $parameters = @{
+            'FilePath' = $programFile
+            'ArgumentList' = $prerequisites[$program]['install_args']
+            'Wait' = $true
+            'PassThru' = $true
+        }
+        if($programFile.EndsWith('.msi')) {
+            $parameters['FilePath'] = 'msiexec.exe'
+            $parameters['ArgumentList'] += @("/i", $programFile)
+        }
+        Write-Output "Installing $programFile"
+        $p = Start-Process @parameters
+        if($p.ExitCode -ne 0) {
+            Throw "Failed to install prerequisite $programFile during the environment setup"
+        }
+    }
+    # Add all the tools to PATH
+    $toolsDirs = @("$GIT_DIR\cmd", "$GIT_DIR\bin", "$7ZIP_DIR")
+    $env:PATH += ';' + ($toolsDirs -join ';')
+}
+
+function Get-LatestCommitID {
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestAdminRouterCommit {
+    # The following id is from https://www.apachelounge.com/download/VC15/binaries/httpd-2.4.33-win64-VC15.zip.txt
+    $commitId = "65525690B713E266A7EA44D60C0D2BCEE4716B2D"
```

Comment:
Looking at https://www.apachelounge.com/download/VC15/binaries/httpd-2.4.33-win64-VC15.zip.txt, isn't this the SHA1 checksum for the zip file (in case people want to check the file integrity) ?

I've even cloned the repository and I couldn't checkout a commit with:

`git reset --hard 65525690b713e266a7ea44d60c0d2bcee4716b2d`

I don't think this is the hash of a commit.

Reply:
I think the confusion was cause by the fact that reused the commitId field for storing $WINDOWS_APACHEL_HTTP_SERVER_SHA256 and mixed it with commit id from the httpd repo.

i will make sure only $WINDOWS_APACHEL_HTTP_SERVER_SHA256 is used to make it consistent

Reply:
It feels wierd to set the lobal `LATEST_COMMIT_ID` to anything else than a commit id of a repository.